maven-core/src/main/java/org/apache/maven/internal/transformation/TransformedArtifact.java

```diff
@@ -38,7 +38,11 @@
  */
 abstract class TransformedArtifact extends DefaultArtifact {
 
-    private final OnChangeTransformer onChangeTransformer;
+    interface Transformer {
+        Path transform() throws Exception;
```

Comment:
I'm not yet convinced that a full java.lang.Exception is desirable here. Possibly an IOException if it's writing/reading files. Possibly no exception at all should be allowed.

Reply:
You're looking at old code.  I've removed `throws Exception` clauses from this package.